### ISSUE TITLE
Contracts that self destruct in their constructors have no contract code published and cannot be verified.

### ISSUE THREAD

**Issue by nisarg2023** [blockscout/blockscout#8272]
now in a change in my node rpc and I successfully get an internal transaction
but no I get to face a new problem verifying the contract

Contracts that self destruct in their constructors have no contract code published and cannot be verified.
Displaying the init data provided of the creating transaction.
![Screenshot from 2023-08-22 18-05-14](https://github.com/blockscout/blockscout/assets/125979189/69cb006f-29ce-4a88-aba0-47762d7b7cc8)

![Screenshot from 2023-08-22 18-06-47](https://github.com/blockscout/blockscout/assets/125979189/d59c6b15-ca43-4640-a7fa-18cb8357deeb)

**Comment by vbaranov** [comment]
@nisarg2023 did you configure smart-contract verification service https://docs.blockscout.com/for-developers/information-and-settings/smart-contract-verification and connect it to your Blockscout instance?

**Comment by nisarg2023** [comment]
@vbaranov Yes i configured the smart-contract verification service and connected it with my Blockscout instance and also verified the contract with api-doc secussifully 
![Screenshot from 2023-08-23 19-12-33](https://github.com/blockscout/blockscout/assets/125979189/d73a705f-e01a-49e9-93a7-bcfa63634b43)
 
but when I try to verify with hardhat I get "contract has been verified but actually it is not verified"

![Screenshot from 2023-08-23 19-15-03](https://github.com/blockscout/blockscout/assets/125979189/b82a6057-03c2-45eb-bf7b-e53782e67424)

**Comment by nisarg2023** [comment]
I am waiting for your response

**Comment by nisarg2023** [comment]
btu when i comment || Chain.get_address_verified_twin_contract(address_hash).verified_contract  this line it work properly


![image](https://github.com/blockscout/blockscout/assets/125979189/fae77eb1-7191-4866-9d64-e607144eea77)

so it can affect another functionality?
